SPECS/kubernetes/version-file-1.18.14.sh

```diff
@@ -0,0 +1,5 @@
+KUBE_GIT_COMMIT='1adde27c8a2253fde5290a6874e9cdb6cb318fc0'
```

Comment:
What do you think about keeping one template script for which we'd replace the commit and version info inside the spec before it's passed to the build? That way we wouldn't need to keep track of these files (add new ones, remove the old) and update everything in one place - the spec file?

It should simplify the updates automation a lot as well.

Reply:
Alternatively, we could use something like this in the spec and skip checking in the script completely:

```
cat > version-file.sh << EOF
KUBE_GIT_COMMIT='%{source_commit_hash}'
KUBE_GIT_TREE_STATE='archive'
KUBE_GIT_VERSION='v%{version}-%{hotfix_number}'
KUBE_GIT_MAJOR='%{version_major}'
KUBE_GIT_MINOR='%{version_minor}+'
EOF
```

The placeholders would be regular RPM macros defined in the spec.

Reply:
by using a template we kind of infer the versioning rules from the tarball name. The real source of trust is version.sh (in k8s code) not what we think the version should be. So I prefer have the automation pipeline creating version file per k8s version we build.
The automation pipeline will have to use version.sh to create those files.

Reply:
Discussed offline, makes sense, closing comment.